### ISSUE TITLE
`react/no-arrow-function-lifecycle` creating invalid code

### ISSUE THREAD

**Issue by devinrhode2** [jsx-eslint/eslint-plugin-react#3337]
Please see this comment: https://github.com/prettier/eslint-plugin-prettier/issues/65#issuecomment-1184764958

**Comment by devinrhode2** [comment]
input:
```tsx
export default class Root extends Component {
  getInitialState = () => ({
    errorImporting: null,
    errorParsing: null,
    errorUploading: null,
    file: null,
    fromExtension: false,
    importSuccess: false,
    isImporting: false,
    isParsing: false,
    isUploading: false,
    parsedResults: null,
    showLongRunningMessage: false,
  });
}
```
invalid output:
```tsx
export default class Root extends Component {
  getInitialState() { return {
    errorImporting: null,
    errorParsing: null,
    errorUploading: null,
    file: null,
    fromExtension: false,
    importSuccess: false,
    isImporting: false,
    isParsing: false,
    isUploading: false,
    parsedResults: null,
    showLongRunningMessage: false,
  }; }); // Extra trailing paren
}
```

**Comment by devinrhode2** [comment]
THANK YOU @ljharb this was driving me bonkers, I was convinced it was a bug related to prettier/eslint integration, but it seems isolated to eslint.

**Comment by ljharb** [comment]
The solution I came up with is a bit quick and dirty, so it probably won't work reliably for atypical forms of semicolon usage, but this should be a start, and will fix the reported issue at least :-)
